Failed to build for wasm32-unknown-unknown with no_std

Hi,

I was playing with plonky2 for `wasm32-unknown-unknown` target without default features and found out that `plonky2_maybe_rayon` erroneously pulls `std` in.

Here is an example reproducing the issue.
<details><summary>Cargo.toml</summary>
<p>

```toml
[package]
name = "plonky2-mayberayon"
version = "0.1.0"
edition = "2021"

[dependencies]
sp-io = { version = "30.0.0", default-features = false }
plonky2_maybe_rayon = { git = "https://github.com/0xPolygonZero/plonky2.git", default-features = false }

[features]
default = ["std"]
std = ["sp-io/std"]
```

</p>
</details>

<details><summary>src/lib.rs</summary>
<p>

```rust
#![no_std]

extern crate sp_io;
extern crate plonky2_maybe_rayon;

```

</p>
</details>

```shell
cargo b --target=wasm32-unknown-unknown --no-default-features

...
   Compiling plonky2-mayberayon v0.1.0 (/home/user/devel/gear-tech/plonky2-mayberayon)
error[E0152]: duplicate lang item in crate `std` (which `plonky2_maybe_rayon` depends on): `panic_impl`.
  |
  = note: the lang item is first defined in crate `sp_io` (which `plonky2_mayberayon` depends on)
  = note: first definition in `sp_io` loaded from /home/user/devel/gear-tech/plonky2-mayberayon/target/wasm32-unknown-unknown/debug/deps/libsp_io-27b44a1938535aab.rmeta
  = note: second definition in `std` loaded from /home/user/.rustup/toolchains/nightly-2023-04-25-x86_64-unknown-linux-gnu/lib/rustlib/wasm32-unknown-unknown/lib/libstd-bdf48c82ab858a37.rlib

For more information about this error, try `rustc --explain E0152`.
error: could not compile `plonky2-mayberayon` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

--- a/maybe_rayon/src/lib.rs
+++ b/maybe_rayon/src/lib.rs
@@ -1,5 +1,6 @@
 #![cfg_attr(not(std), no_std)]
 
+#[cfg(not(feature = "parallel"))]
 extern crate alloc;
 
 #[cfg(feature = "parallel")]

--- a/maybe_rayon/src/lib.rs
+++ b/maybe_rayon/src/lib.rs
@@ -1,7 +1,14 @@
+#![cfg_attr(not(std), no_std)]
+
+extern crate alloc;
+
 #[cfg(not(feature = "parallel"))]
-use core::{
-    iter::{FlatMap, IntoIterator, Iterator},
-    slice::{Chunks, ChunksExact, ChunksExactMut, ChunksMut},
+use {
+    core::{
+        iter::{FlatMap, IntoIterator, Iterator},
+        slice::{self, Chunks, ChunksExact, ChunksExactMut, ChunksMut},
+    },
+    alloc::vec::Vec,
 };
 
 #[cfg(feature = "parallel")]
@@ -53,7 +60,7 @@ where
 #[cfg(not(feature = "parallel"))]
 impl<'data, T: 'data> MaybeParIter<'data> for Vec<T> {
     type Item = &'data T;
-    type Iter = std::slice::Iter<'data, T>;
+    type Iter = slice::Iter<'data, T>;
 
     fn par_iter(&'data self) -> Self::Iter {
         self.iter()
@@ -63,7 +70,7 @@ impl<'data, T: 'data> MaybeParIter<'data> for Vec<T> {
 #[cfg(not(feature = "parallel"))]
 impl<'data, T: 'data> MaybeParIter<'data> for [T] {
     type Item = &'data T;
-    type Iter = std::slice::Iter<'data, T>;
+    type Iter = slice::Iter<'data, T>;
 
     fn par_iter(&'data self) -> Self::Iter {
         self.iter()
@@ -102,7 +109,7 @@ where
 #[cfg(not(feature = "parallel"))]
 impl<'data, T: 'data> MaybeParIterMut<'data> for Vec<T> {
     type Item = &'data mut T;
-    type Iter = std::slice::IterMut<'data, T>;
+    type Iter = slice::IterMut<'data, T>;
 
     fn par_iter_mut(&'data mut self) -> Self::Iter {
         self.iter_mut()
@@ -112,7 +119,7 @@ impl<'data, T: 'data> MaybeParIterMut<'data> for Vec<T> {
 #[cfg(not(feature = "parallel"))]
 impl<'data, T: 'data> MaybeParIterMut<'data> for [T] {
     type Item = &'data mut T;
-    type Iter = std::slice::IterMut<'data, T>;
+    type Iter = slice::IterMut<'data, T>;
 
     fn par_iter_mut(&'data mut self) -> Self::Iter {
         self.iter_mut()

--- a/maybe_rayon/src/lib.rs
+++ b/maybe_rayon/src/lib.rs
@@ -2,15 +2,6 @@
 
 extern crate alloc;
 
-#[cfg(not(feature = "parallel"))]
-use {
-    core::{
-        iter::{FlatMap, IntoIterator, Iterator},
-        slice::{self, Chunks, ChunksExact, ChunksExactMut, ChunksMut},
-    },
-    alloc::vec::Vec,
-};
-
 #[cfg(feature = "parallel")]
 pub use rayon::{
     self,
@@ -25,6 +16,14 @@ use rayon::{
     slice::{
         Chunks as ParChunks, ChunksExact as ParChunksExact, ChunksExactMut as ParChunksExactMut,
         ChunksMut as ParChunksMut, ParallelSlice, ParallelSliceMut,
+    },
+};
+#[cfg(not(feature = "parallel"))]
+use {
+    alloc::vec::Vec,
+    core::{
+        iter::{FlatMap, IntoIterator, Iterator},
+        slice::{self, Chunks, ChunksExact, ChunksExactMut, ChunksMut},
     },
 };
 

--- a/plonky2/Cargo.toml
+++ b/plonky2/Cargo.toml
@@ -31,7 +31,7 @@ plonky2_util = { path = "../util", default-features = false }
 rand = { version = "0.8.4", default-features = false }
 rand_chacha = { version = "0.3.1", optional = true, default-features = false }
 serde = { version = "1.0", default-features = false, features = ["derive", "rc"] }
-serde_json = "1.0"
+serde_json = { version = "1.0", default-features = false, features = ["alloc"] }
 static_assertions = { version = "1.1.0", default-features = false }
 unroll = { version = "0.1.5", default-features = false }
 web-time = { version = "1.0.0", optional = true }

--- a/plonky2/Cargo.toml
+++ b/plonky2/Cargo.toml
@@ -31,7 +31,6 @@ plonky2_util = { path = "../util", default-features = false }
 rand = { version = "0.8.4", default-features = false }
 rand_chacha = { version = "0.3.1", optional = true, default-features = false }
 serde = { version = "1.0", default-features = false, features = ["derive", "rc"] }
-serde_json = { version = "1.0", default-features = false, features = ["alloc"] }
 static_assertions = { version = "1.1.0", default-features = false }
 unroll = { version = "0.1.5", default-features = false }
 web-time = { version = "1.0.0", optional = true }
@@ -46,6 +45,7 @@ num_cpus = { version = "1.14.0", default-features = false }
 rand = { version = "0.8.4", default-features = false, features = ["getrandom"] }
 rand_chacha = { version = "0.3.1", default-features = false }
 serde_cbor = { version = "0.11.2" }
+serde_json = { version = "1.0" }
 structopt = { version = "0.3.26", default-features = false }
 tynm = { version = "0.1.6", default-features = false }
 

